app/brave_main_delegate.cc

```diff
@@ -235,3 +248,40 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
 
   return ret;
 }
+
+#if defined(OS_ANDROID)
+void BraveMainDelegate::AdjustSyncServiceUrlForAndroid(
+    std::string* brave_sync_service_url) {
+  DCHECK_NE(brave_sync_service_url, nullptr);
+  const char kProcessTypeHost[] = "host";
+  const char kProcessTypeSwitchName[] = "type";
+
+  // On Android we can detect data dir only on host process, and we cannot
+  // for example on renderer or gpu-process, because JNI is not initialized
+  // And no sense to override sync service url for them in anyway
+  std::string process_type = kProcessTypeHost;
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          kProcessTypeSwitchName)) {
+    process_type = base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+        kProcessTypeSwitchName);
+  }
+
+  if (process_type != kProcessTypeHost) {
+    return;
+  }
+
+  base::FilePath data_directory;
+  bool get_dir_result = base::android::GetDataDirectory(&data_directory);
+  if (!get_dir_result) {
+    return;
+  }
+
+  const char kStagingSyncServerMarkerName[] = "use_staging_sync_server";
+  base::FilePath staging_marker_path =
```

Comment:
can't we set a pref through jni and check here instead of using marker file? We only need that pref for browser process

Reply:
yes, we can, this will require to have one extra Java class with JNI support and call generated object in this place. I tried to avoid this for a single flag, but will will change.